packages/hash/api/src/graphql/resolvers/paginationConnection/pageSearchResultConnection.ts

```diff
@@ -0,0 +1,144 @@
+    ({ hits, cursor, pageCount } = await search.continuePaginatedSearch({
+      cursor: after,
+    }));
+  } else if (query && pageSize) {
+    ({ hits, cursor, pageCount } = await search.startPaginatedSearch({
+      pageSize,
+      index: ENTITIES_SEARCH_INDEX,
+      fields: {
+        [ENTITIES_SEARCH_FIELD]: {
+          query,
+          fuzziness: "AUTO",
+          operator: "or",
+          presence: "must",
+        },
+        // Only fetch entities with the "Text" system type.
+        // These will contain a "belongsToParent" property
+        entityTypeId: {
+          query: textType.entityId,
+          // Fuzziness is set to 0, such that an ID that is lexicographically similar don't match
+          // This filter used to be done in resolvers.
+          fuzziness: 0,
+          operator: "and",
+          presence: "must",
+        },
+        "belongsToPage.accountId": {
+          query: accountId,
+          // Fuzziness is set to 0, such that an ID that is lexicographically similar don't match
+          // This filter used to be done in resolvers.
+          fuzziness: 0,
+          operator: "and",
+          presence: "must",
+        },
+      },
+    }));
+  } else {
+    throw new UserInputError(
+      "Could not execute search. Please revise arguments.",
+    );
+  }
+
+  // For all text entity matches, find the pages and the blocks within those pages where
+  // the text match is present. Include hits corresponding only to the latest version
+  // of a page.
+  // Note: we filter the resulting array to keep only those pages which match the
+  // `accountId` set in the query. A page may link to entities which belong to a
+  // different account to that of the page. We don't filter by the `accountId` before this
+  // point as it would remove these pages with cross-account links.
+
+  // @todo: filtering already happens in the search index, this filtering should be redundant.
+  // it is used for type assertion only.
+  const textMatches = hits
+    .filter(
+      (hit): hit is TextSearchHit =>
+        hit.document.entityTypeId === textType.entityId,
+    )
+    .map(
+      (it) =>
+        <PageSearchResult>{
+          score: it.score,
+          page: it.document.belongsToPage!,
+          /**
+           * @todo: Currently we are not getting the (parent) block of a text system type in the search-loader
+           * This means that scrolling to a block after selecting a search hit is not possible even though
+           * the API implies it is. We will need to implement indexing of the "parent block" within the search-loader.
+           */
+          block: undefined,
+          text: {
+            accountId: it.document.accountId,
+            entityId: it.document.entityId,
+            entityVersionId: it.document.entityTypeVersionId,
+          },
+          content: it.document.fullTextSearch || "",
+        },
+    );
+
+  textMatches.sort((matchA, matchB) => matchB.score - matchA.score);
```

Comment:
Nit pick: `.sort` does not need to happen on a separate line, and we can destructure the args passed to the mapping function (instead of using `it`, which is a typo for `hit`?):
```suggestion
  const textMatches = hits
    .filter(
      (hit): hit is TextSearchHit =>
        hit.document.entityTypeId === textType.entityId,
    )
    .map<PageSearchResult>(({ score, document }) => ({
      score,
      page: document.belongsToPage!,
      /**
       * @todo: Currently we are not getting the (parent) block of a text system type in the search-loader
       * This means that scrolling to a block after selecting a search hit is not possible even though
       * the API implies it is. We will need to implement indexing of the "parent block" within the search-loader.
       */
      block: undefined,
      text: {
        accountId: document.accountId,
        entityId: document.entityId,
        entityVersionId: document.entityTypeVersionId,
      },
      content: document.fullTextSearch || "",
    }))
    .sort((matchA, matchB) => matchB.score - matchA.score);
```

Reply:
Yes, good point. I left it separate as it is a mutating operation, but I don't have strong feelings about having it be a statement for itself :)